doc/devdocs/development/new-powertoy.md

```diff
@@ -0,0 +1,287 @@
+# 🧭 Creating a New PowerToy: End-to-End Developer Guide
+
+First of all, thank you for wanting to contribute to PowerToys. The work we do would not be possible without the support of community supporters like you.
+
+This guide documents the process of building a new PowerToys utility from scratch, including architecture decisions, integration steps, and common pitfalls.
+
+---
+
+## 1. Overview and Prerequisites
+
+A PowerToy module is a self-contained utility integrated into the PowerToys ecosystem. It can be UI-based, service-based, or both.
+
+### Requirements
+
+* [Visual Studio 2022](https://visualstudio.microsoft.com/downloads/) and the following workloads/individual components:
+  - Desktop Development with C++
+  - WinUI application development
+  - .NET desktop development
+  - Windows 10 SDK (10.0.22621.0)
+  - Windows 11 SDK (10.0.26100.3916)
+* .NET 8 SDK
+* Clone the [PowerToys repository](https://github.com/microsoft/PowerToys/tree/main) locally
+* [Validate that you are able to build and run](https://github.com/microsoft/PowerToys/blob/main/doc/devdocs/development/debugging.md) `PowerToys.slnx`.
+
+Optional: 
+* [WiX v5 toolset](https://github.com/microsoft/PowerToys/tree/main) for the installer
+
+> Note: To ensure all the correct VS Workloads are installed, use [the WinGet configuration files](https://github.com/microsoft/PowerToys/tree/e13d6a78aafbcf32a4bb5f8581d041e1d057c3f1/.config) in the project repository. (Use the one that matches your VS distribution. ie: VS Community would use `configuration.winget`)
+
+### Folder Structure
+
+```
+src/
+  modules/
+    your_module/
+      YourModule.sln
+      YourModuleInterface/
+      YourModuleUI/ (if needed)
+      YourModuleService/ (if needed)
+```
+
+---
+## 2. Design and Planning
+
+### Decide the Type of Module
+Think about how your module works and which existing modules behave similarly. You are going to want to think about the UI needed for the application, the lifecycle, whether it is a service that is always running or event based. Below are some basic scenarios with some modules to explore. You can write your application in C++ or C#.
+* **UI-only:** e.g., ColorPicker
```

Comment:
General: personal preference: can we use `-` dash for lists? Just to prevent any issue with all the `*`s together 🤓